### PR TITLE
RID signaling

### PR DESCRIPTION
**Pull request by bbaldino**
depends on https://github.com/jitsi/jitsi/pull/372#pullrequestreview-48758692

### CODE REVIEW

**Comment by bbaldino** [comment]
don't merge this until the jitsi commit goes in and i can update the version

File: pom.xml [hunk 1]
```diff
@@ -103,7 +103,7 @@
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>jitsi-protocol-jabber</artifactId>
-      <version>2.11-20170516.192102-5</version>
+      <version>2.12-SNAPSHOT</version>
```
Comment (bbaldino) [inline pom.xml]: shit, this was accidental.

**Comment by paweldomas** [comment]
Can you get rid of trash commits about moving back and forth to java 8 ?

File: src/main/java/org/jitsi/impl/protocol/xmpp/colibri/ColibriConferenceImpl.java [hunk 2]
```diff
@@ -614,8 +614,8 @@ public void updateTransportInfo(
      * {@inheritDoc}
      */
     @Override
-    public void updateSourcesInfo(MediaSSRCMap ssrcs,
-                                  MediaSSRCGroupMap ssrcGroups,
+    public void updateSourcesInfo(MediaSourceMap ssrcs,
```
Comment (paweldomas) [inline src/main/java/org/jitsi/impl/protocol/xmpp/colibri/ColibriConferenceImpl.java]: I think we need to rename the code fully by also renaming variables from SSRCs to sources

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java [hunk 3]
```diff
@@ -1431,14 +1431,14 @@ public XMPPError onRemoveSource(JingleSession sourceJingleSession,
      *
      * @param sourceJingleSession source Jingle session from which SSRCs are
      *                            being removed.
-     * @param ssrcsToRemove the {@link MediaSSRCMap} of SSRCs to be removed from
+     * @param ssrcsToRemove the {@link MediaSourceMap} of SSRCs to be removed from
      *                      the conference.
      * @param updateChannels tells whether or not SSRC update request should be
      *                       sent to the bridge.
      */
     private void removeSSRCs(JingleSession        sourceJingleSession,
```
Comment (paweldomas) [inline src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java]: please rename this method to removeSources and any mentions of SSRCs in JavaDoc

Please take a look on other modified methods as well

File: src/main/java/org/jitsi/protocol/xmpp/AbstractOperationSetJingle.java [hunk 4]
```diff
@@ -294,6 +294,9 @@ protected void processJingleIQ(JingleIQ iq)
         switch (action)
         {
         case SESSION_ACCEPT:
+            logger.error("BB: Received session accept: \n" + iq.toXML());
```
Comment (paweldomas) [inline src/main/java/org/jitsi/protocol/xmpp/AbstractOperationSetJingle.java]: is it here by mistake ?

File: src/test/java/org/jitsi/jicofo/SourceUtil.java [hunk 5]
```diff
@@ -28,31 +28,46 @@
  *
  * @author Pawel Domas
  */
-public class SSRCUtil
+public class SourceUtil
 {
-    static public SourcePacketExtension createSSRC(long ssrc, String[][] params)
+    static public SourcePacketExtension createSourceWithSsrc(long ssrc, String[][] params)
```
Comment (paweldomas) [inline src/test/java/org/jitsi/jicofo/SourceUtil.java]: createSourceWithSSRC would be preferred

File: src/test/java/org/jitsi/jicofo/SourceUtil.java [hunk 6]
```diff
@@ -28,31 +28,46 @@
...
     {
         SourcePacketExtension ssrcPE = new SourcePacketExtension();
 
         ssrcPE.setSSRC(ssrc);
+        setSourceParams(ssrcPE, params);
 
+        return ssrcPE;
+    }
+
+    static public SourcePacketExtension createSourceWithRid(String rid, String[][] params)
```
Comment (paweldomas) [inline src/test/java/org/jitsi/jicofo/SourceUtil.java]: createSourceWithRID would be preferred

File: src/main/java/org/jitsi/jicofo/SSRCValidator.java [hunk 7]
```diff
@@ -53,135 +53,140 @@
...
      */
-    private final MediaSSRCGroupMap ssrcGroups;
+    private final MediaSourceGroupMap sourceGroups;
 
     /**
-     * The limit SSRCs count per media type allowed to be stored by
+     * The limit sources count per media type allowed to be stored by
      * the {@link Participant} at a time.
+     * NOTE(bbaldino): changing this to max source count to have it
+     * apply to rids as well, not sure if that's appropriate
```
Comment (paweldomas) [inline src/main/java/org/jitsi/jicofo/SSRCValidator.java]: RIDs would be preferred

File: src/test/java/org/jitsi/jicofo/MediaSourceGroupMapTest.java [hunk 8]
```diff
@@ -30,137 +30,168 @@
...
 @RunWith(JUnit4.class)
-public class MediaSSRCGroupMapTest
+public class MediaSourceGroupMapTest
 {
+    @Test
+    public void testContainsGroup()
+    {
+        /**
+         * Make sure that groups with the same semantic but different source
+         * types (rid vs ssrc) don't collide
```
Comment (paweldomas) [inline src/test/java/org/jitsi/jicofo/MediaSourceGroupMapTest.java]: RID vs SSRC

File: src/test/java/org/jitsi/jicofo/MediaSourceGroupMapTest.java [hunk 9]
```diff
@@ -30,137 +30,168 @@
...
+                SourceGroupPacketExtension.SEMANTICS_SIMULCAST,
+                new SourcePacketExtension[] {
+                        SourceUtil.createSourceWithSsrc(123123L, new String[][]{}),
+                        SourceUtil.createSourceWithSsrc(456456L, new String[][]{}),
+                        SourceUtil.createSourceWithSsrc(789789L, new String[][]{})
+
+                }
+        );
+        sourceGroups.addSourceGroup("video", ssrcSimGroup);
+        // Now create an rid group
```
Comment (paweldomas) [inline src/test/java/org/jitsi/jicofo/MediaSourceGroupMapTest.java]: RID group

File: src/test/java/org/jitsi/jicofo/MediaSourceMapTest.java [hunk 10]
```diff
@@ -35,69 +35,104 @@
...
 @RunWith(JUnit4.class)
-public class MediaSSRCMapTest
+public class MediaSourceMapTest
 {
+
+    @Test
+    public void testGetSourcesSsrcAndRid()
+    {
+        /**
+         * Test to make sure both ssrc and rid sources are extracted
```
Comment (paweldomas) [inline src/test/java/org/jitsi/jicofo/MediaSourceMapTest.java]: 'both SSRC and RID sources'

**Comment by paweldomas** [comment]
about Ssrc and Rid vs SSRC and RID feel free to disregard my comments

File: src/main/java/org/jitsi/impl/protocol/xmpp/colibri/ColibriConferenceImpl.java (hunk 2 above)
Comment (bbaldino) [inline src/main/java/org/jitsi/impl/protocol/xmpp/colibri/ColibriConferenceImpl.java]: yes you're right, i did this in most places but looks like i missed this one.  will fix it.

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java (hunk 3 above)
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java]: fixed, and will do another scan for uses of ssrc

File: src/main/java/org/jitsi/jicofo/SSRCValidator.java (hunk 7 above)
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/SSRCValidator.java]: gonna get rid of this note, i had it as more of a TODO to check with you.  do you think it's alright to have it apply to ssrc + rids?  i don't think it makes sense to just be one or the other, but currently there will be duplication (each stream from firefox will have an ssrc and an rid souce, so there will be 2x normal...i don't know what we really rely on this for though)

File: src/test/java/org/jitsi/jicofo/MediaSourceGroupMapTest.java (hunk 8 above)
Comment (bbaldino) [inline src/test/java/org/jitsi/jicofo/MediaSourceGroupMapTest.java]: fixed (i left the other capitalization scheme for the variable/function names, but i agree this capitalization is appropriate for comments)

File: src/test/java/org/jitsi/jicofo/MediaSourceMapTest.java (hunk 10 above)
Comment (bbaldino) [inline src/test/java/org/jitsi/jicofo/MediaSourceMapTest.java]: fixed

**Comment by bbaldino** [comment]
@paweldomas thanks for the comments!  with regards to the commit/revert for java 8...we should squash this when merging anyway so it shouldn't cloud up the history

**Comment by paweldomas** [comment]
PR tests will not work until jitsi PR is merged and pom.xml updated here.

**Pull request merged**